components/ml/org.wso2.carbon.ml.rest.api/src/main/java/org/wso2/carbon/ml/rest/api/ModelApiV10.java

```diff
@@ -127,6 +127,25 @@ public Response buildModel(@PathParam("modelId") long modelId) {
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build();
         }
     }
+
+    @POST
+    @Path("/{modelId}/publish")
+    @Produces("application/json")
+    @Consumes("application/json")
+    public Response publishModel(@PathParam("modelId") long modelId) {
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        int tenantId = carbonContext.getTenantId();
+        String userName = carbonContext.getUsername();
+        try {
+            mlModelHandler.publishModel(tenantId, userName, modelId);
+            return Response.ok().build();
+        } catch (Exception e) {
```

Comment:
publishModel method in MLModelHandler throws  MLModelPublisherException. Is there any reason to catch the exception supper object here ?